pkg/skaffold/deploy/kpt/kpt.go

```diff
@@ -103,30 +104,43 @@ func versionCheck(dir string) error {
 	// Users can choose not to use kustomize in kpt deployer mode. We only check the kustomize
 	// version when kustomization.yaml config is directed under .deploy.kpt.dir path.
 	_, err = kustomize.FindKustomizationConfig(dir)
-	if err == nil {
-		kustomizeCmd := exec.Command("kustomize", "version")
-		out, err := util.RunCmdOut(kustomizeCmd)
-		if err != nil {
-			return fmt.Errorf("kustomize is not installed yet\nSee kpt installation: %v",
-				kustomizeDownloadLink)
-		}
-		versionInfo := strings.TrimSuffix(string(out), "\n")
-		// Kustomize version information is in the form of
-		// {Version:$VERSION GitCommit:$COMMIT BuildDate:1970-01-01T00:00:00Z GoOs:darwin GoArch:amd64}
-		re := regexp.MustCompile(kustomizeVersionRegexP)
-		match := re.FindStringSubmatch(versionInfo)
-		if len(match) != 2 {
-			return fmt.Errorf("unknown kustomize version %v\nPlease upgrade your "+
+	if err != nil {
+		return nil
+	}
+
+	kustomizeCmd := exec.Command("kustomize", "version")
+	out, err = util.RunCmdOut(kustomizeCmd)
+	if err != nil {
+		return fmt.Errorf(
+			"kustomize is not installed yet\nSee kpt installation: %v",
+			kustomizeDownloadLink)
+	}
+
+	versionInfo := strings.TrimSuffix(string(out), "\n")
+	// Kustomize version information is in the form of
+	// {Version:$VERSION GitCommit:$COMMIT BuildDate:1970-01-01T00:00:00Z GoOs:darwin GoArch:amd64}
+	re := regexp.MustCompile(kustomizeVersionRegexP)
+	match := re.FindStringSubmatch(versionInfo)
+	if len(match) != 3 {
+		fmt.Printf(
+			"unknown kustomize version. \nPlease upgrade your "+
 				"local kustomize CLI to a version >= %v\nSee kustomize installation: %v",
-				string(out), kustomizeMinVersion, kustomizeDownloadLink)
-		}
-		if !semver.IsValid(match[1]) || semver.Compare(match[1], kustomizeMinVersion) < 0 {
-			return fmt.Errorf("you are using kustomize %q\n"+
+			kustomizeMinVersion, kustomizeDownloadLink)
+	} else {
```

Comment:
nit: this else is not necessaroy.

Reply:
That would cause multiple messages to be printed in the case of a version which doesn't provide us with 3 match groups. This else makes sure that only the 'unknown version' message is being output in that situation.